internal/http/services/owncloud/ocdav/ocdav_blackbox_test.go

```diff
@@ -120,13 +134,383 @@ var _ = Describe("ocdav", func() {
+
+	// basic metadata
+	// PROPFIND
+	// MKCOL
+	// DELETE
+
+	// basic data
+	// PUT
+	// GET
+	// HEAD
+
+	// move & copy
+	// MOVE
+	// COPY
+
+	// additional methods
+	// PROPPATCH
+	// LOCK
+	// UNLOCK
+	// REPORT
+	// POST (Tus)
+	// OPTIONS?
+
+	Context("at the very legacy /webdav endpoint", func() {
+
+		BeforeEach(func() {
+			// set the webdav endpoint to test
+			basePath = "/webdav"
+
+			// path based requests at the /webdav endpoint first look up the storage space
+			client.On("ListStorageSpaces", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.ListStorageSpacesRequest) bool {
+				p := string(req.Opaque.Map["path"].Value)
+				return p == "/" || strings.HasPrefix(p, "/users")
+			})).Return(&cs3storageprovider.ListStorageSpacesResponse{
+				Status:        status.NewOK(ctx),
+				StorageSpaces: []*cs3storageprovider.StorageSpace{userspace},
+			}, nil)
+		})
+
+		Describe("PROPFIND", func() {
+
+			var rr *httptest.ResponseRecorder
+			var req *http.Request
+			var err error
+
+			BeforeEach(func() {
+				// setup the request
+				rr = httptest.NewRecorder()
+				req, err = http.NewRequest("PROPFIND", basePath, strings.NewReader(""))
+				Expect(err).ToNot(HaveOccurred())
+				req = req.WithContext(ctx)
+
+			})
+			When("the gateway returns a file list", func() {
+				It("returns a multistatus with the file info", func() {
+
+					client.On("Stat", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.StatRequest) bool {
+						return req.Ref.Path == "/users/username"
+					})).Return(&cs3storageprovider.StatResponse{
+						Status: status.NewOK(ctx),
+						Info:   userspace.RootInfo,
+					}, nil)
+
+					handler.Handler().ServeHTTP(rr, req)
+					Expect(rr).To(HaveHTTPStatus(http.StatusMultiStatus))
+					Expect(rr).To(HaveHTTPBody(Not(BeEmpty())), "Body must not be empty")
+					// TODO test listing more thoroughly
+				})
+
+			})
+
+			When("the gateway returns not found", func() {
+				It("returns a not found status", func() {
+
+					client.On("Stat", mock.Anything, mock.MatchedBy(func(req *cs3storageprovider.StatRequest) bool {
+						return req.Ref.Path == "/users/username"
+					})).Return(&cs3storageprovider.StatResponse{
+						Status: status.NewNotFound(ctx, "resource not found"),
+					}, nil)
+
+					handler.Handler().ServeHTTP(rr, req)
+					Expect(rr).To(HaveHTTPStatus(http.StatusMultiStatus))
+					Expect(rr).To(HaveHTTPBody("<d:multistatus xmlns:s=\"http://sabredav.org/ns\" xmlns:d=\"DAV:\" xmlns:oc=\"http://owncloud.org/ns\"><d:response><d:href>/webdav</d:href><d:propstat><d:prop><oc:name>username</oc:name><d:resourcetype></d:resourcetype><d:getcontentlength>0</d:getcontentlength><oc:favorite>0</oc:favorite></d:prop><d:status>HTTP/1.1 200 OK</d:status></d:propstat><d:propstat><d:prop><oc:file-parent></oc:file-parent></d:prop><d:status>HTTP/1.1 404 Not Found</d:status></d:propstat></d:response></d:multistatus>"), "Body must not be empty")
```

Comment:
I wonder how static the response xml is. Maybe we could just do something like `ContainSubstring("<d:status>HTTP/1.1 404 Not Found</d:status>")` instead to prevent false failures?

Reply:
I'll use sth like this:
```

					Expect(rr).To(HaveHTTPBody(
						And(
							ContainSubstring("<s:exception>Sabre\\DAV\\Exception\\MethodNotAllowed</s:exception>"),
							ContainSubstring("<s:message>The resource you tried to create already exists</s:message>"))),
						"Body must contain sabredav exception and message")
```

Reply:
and the d:status is part of a propfind multistatus response xml, which is different then the http status of the response...